helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/SoftConstraintWeightModel.java

```diff
@@ -0,0 +1,47 @@
+package org.apache.helix.controller.rebalancer.waged.constraints;
+
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * The class retrieves the offline model that defines the relative importance of soft constraints.
+ */
+public class SoftConstraintWeightModel {
+  private float MIN_SCORE = -1000.0f;
+  private float MAX_SCORE = 1000.0f;
+  private static Map<SoftConstraint.Type, Float> MODEL;
+
+  static {
+    // TODO either define the weights in property files or zookeeper node or static human input
+    MODEL = ImmutableMap.<SoftConstraint.Type, Float> builder()
+        .put(SoftConstraint.Type.LEAST_MOVEMENTS, 1.0f)
+        .put(SoftConstraint.Type.LEAST_PARTITION_COUNT, 1.0f)
+        .put(SoftConstraint.Type.LEAST_USED_NODE, 1.0f).build();
+  }
+
+  interface ScoreScaler {
+    /**
+     * Method to scale the origin score to a normalized range
+     * @param originScore The origin score of a range
+     * @return The normalized value between 0 - 1
+     */
+    float scale(float originScore);
+  }
+
+  private ScoreScaler MIN_MAX_SCALER =
+      originScore -> (originScore - MIN_SCORE) / (MAX_SCORE - MIN_SCORE);
+
+  /**
+   * Given the calculated scores map by soft constraints, get the sum of scores
+   * @param originScoresMap The origin scores of soft constraints
+   * @return The sum of double type
+   */
+  public double sumOfScores(Map<SoftConstraint, Float> originScoresMap) {
+    return originScoresMap.entrySet().stream().map(constraintToOriginScore -> {
+      float score = MIN_MAX_SCALER.scale(constraintToOriginScore.getValue());
+      float weight = MODEL.getOrDefault(constraintToOriginScore.getKey().getType(), 0f);
```

Comment:
How to ensure the input map includes all the constraint evaluation result? Shall we throw an exception if not?

Reply:
The MODEL is defined statically and not set dynamically on the fly. I think we should assume all types have a valid value or we're doing a bad job when assigning the weight.  Throw exception will fail all the rest operations I don't think it's good idea